### DISCUSSION TITLE
Issue with Nested Folders in Media Library

### DISCUSSION THREAD

**Discussion by RDjarbeng** [sveltia/sveltia-cms#229] [Q&A] [answered]
I'm currently facing an issue with organizing images in my media folder. For each post, I often add multiple images and, to avoid filename conflicts, I organize them into subfolders. For example, the path to the images would look like this: `assets/images/example_folder/image1.png`.

However, I've noticed that:
1. These folders do not appear when I'm selecting images in the assets section.
2. The nested folders and images also don't show up on the collections page.

Here are some screenshots for reference:
- **Screenshot 1:**
  ![Image 1](https://github.com/user-attachments/assets/b03e1e21-41b9-4574-b72f-0bf42169831e)
- **Screenshot 2 (Zoomed in):**
  ![Image 2](https://github.com/user-attachments/assets/e36bf990-43c6-497f-9101-1a658ac046fa)

My configuration in `config.yml` for images looks like this:

```yaml
media_folder: "assets/images"
public_folder: "/assets/images"
```

You can find the full `config.yml` [here](https://github.com/RDjarbeng/RDjarbeng/blob/main/admin/config.yml).

Currently, in order to use images from these nested folders, I have to manually type the paths in the markdown. While this doesn’t break the build, and the images display correctly on the site, it adds extra manual work.

### My Questions:
- Is there a way to make these image folders appear when selecting assets?
- How can I ensure the images from these folders also show in the post preview when used as the post image?

Any guidance or suggestions would be much appreciated!

**Comment by RDjarbeng** [comment]
I am not sure if this is the same as #227

**Comment by kyoshino** [comment]
#227 is something else: [Nested Collections](https://decapcms.org/docs/collection-nested/).

In Sveltia CMS, it’s possible to [use different media folders for different collections](https://github.com/sveltia/sveltia-cms#using-a-custom-media-folder-for-a-collection) (in your case Posts, Personal and Authors), but nested folders are not supported yet. It’s on my todo list.

I don’t think Netlify/Decap CMS [supports it yet](https://github.com/decaporg/decap-cms/issues/3240) either, but [Static CMS](https://github.com/StaticJsCMS/static-cms), a now-discontinued community fork of Netlify CMS, did. So I plan to provide compatibility before the 1.0 release.

**Reply by kyoshino** [reply]
I’ll see if it’s possible to at least display images stored in nested folders.

The ability to create new folders in the Media Library like Static CMS is another thing.

**Reply by RDjarbeng** [reply]
Alright, thanks for the quick reply. Looking forward to it 🚀

**Comment by kyoshino** [comment]
Shipped [v0.42.1](https://github.com/sveltia/sveltia-cms/releases/tag/v0.42.1) with the fix!

**Comment by RDjarbeng** [comment]
How do I upgrade to v0.42.1? Do I just refresh the page(tried that) ?
`<script src="https://unpkg.com/@sveltia/cms/dist/sveltia-cms.js"></script>`

I still cant' see the folders.

**Reply by RDjarbeng** [reply]
Is there a way to check which version I'm currently using?

**Reply by kyoshino** [reply]
Sorry for the confusion. You just need to reload the CMS page to upgrade.

Folders are not displayed yet, but all assets in subfolders are now available. I’ll reopen this discussion not to forget about the folder navigation/creation support. 🙇🏼 

To check the version, first enable Developer Mode in Settings > Advanced:

<img width="259" alt="image" src="https://github.com/user-attachments/assets/3349cdc2-c76c-41e2-ad69-70863301ab31">

Then click the Help menu:

<img width="272" alt="image" src="https://github.com/user-attachments/assets/8363dfb3-5764-4a5a-8605-247e3c025dbe">

**Reply by RDjarbeng** [reply]
Thanks. 

However, I still have the same behaviour.

![image](https://github.com/user-attachments/assets/1e09fe38-b1c6-43fc-b82f-0d94c06cb31d)

The assets are not displaying when I check the post

![image](https://github.com/user-attachments/assets/451619ed-8233-4cb3-8bae-effcfd71045f)


Using version 0.42.2
![image](https://github.com/user-attachments/assets/c7054535-7b27-4914-a015-ef5821bbe8d7)

**Reply by kyoshino** [reply]
Hmm, will check back.

**Reply by kyoshino** [reply, marked as answer]
The fix landed in the [latest version](https://github.com/sveltia/sveltia-cms/releases/tag/v0.45.0). `/assets/images/weather_post/wunderground.png` should appear in the CMS UI now.

**Comment by RDjarbeng** [comment]
Okay, I have checked and it seems to fix these issues

These folders now appear when I'm selecting images in the assets section. - ✅
The nested folders and images also now show up on the collections page. -✅

Thank you

**Comment by RDjarbeng** [comment]
There’s just one small issue I’ve noticed.

Currently, the image picker doesn’t display nested folders, so all images—whether in the root or in subfolders—appear together in one list. This makes it hard to tell which images are actually in the root folder versus those that are organized in subfolders.

The bigger challenge here is that, without folder separation, the images also don’t seem to be sorted, resulting in a rather random display. As a result, finding a specific image takes more time since I have to sift through the entire list rather than simply navigating to the folder where it’s stored.

**Reply by kyoshino** [reply]
The files are now ordered by name as of [v0.45.3](https://github.com/sveltia/sveltia-cms/releases/tag/v0.45.3). I’ll implement folder navigation asap.

**Comment by RDjarbeng** [comment]
I think I'm going to convert this into an issue so that it's easier to track

**Comment by RDjarbeng** [comment]
Nope, no need issue already open in #301
